src/System.Data.SqlClient/src/System/Data/ProviderBase/DbConnectionFactory.cs

```diff
@@ -125,8 +125,28 @@ virtual internal DbConnectionPoolGroupProviderInfo CreateConnectionPoolGroupProv
 
         private Timer CreatePruningTimer()
         {
-            TimerCallback callback = new TimerCallback(PruneConnectionPoolGroups);
-            return new Timer(callback, null, PruningDueTime, PruningPeriod);
+            Timer timer;
+            // Don't capture the current ExecutionContext and its AsyncLocals onto the timer causing them to live forever
+            bool restoreFlow = false;
+            try
+            {
+                if (!ExecutionContext.IsFlowSuppressed())
```

Comment:
This code/pattern is being used at multiple places. Can this be refactored into a reusable method. AdapterUtil.cs is a good place to put such utilities in System.Data

Reply:
Done